src/linked_hash_map.rs

```diff
@@ -503,6 +503,21 @@ where
             }
         }
     }
+
+    /// Returns the `CursorMut` over the _guard_ node.
+    ///
+    /// Note: The `CursorMut` over the _guard_ node in an empty `LinkedHashMap` will always
+    ///       return `None` as its current eliment, regardless of any move in any direction.
+    pub fn cursor_mut(&mut self) -> CursorMut<K, V, S> {
+        unsafe { ensure_guard_node(&mut self.values) };
+        CursorMut {
+            cur: self.values.as_ptr(),
+            hash_builder: &self.hash_builder,
+            free: &mut self.free,
+            values: &mut self.values,
+            table: &mut self.table,
+        }
+    }
```

Comment:
Indeed, it's a valid point. I too was pondering over the same issue, specifically about the cursor's position when we invoke it on an empty instance of `LinkedHashMap`.

This method is primarily for convenience. I'm open to your suggestions. Could you please share your thoughts on the appropriate behavior for an empty map?

Reply:
It would point to the guard node in the case of an empty map, I'm proposing that we mimic https://doc.rust-lang.org/stable/std/collections/struct.LinkedList.html#method.cursor_back_mut and https://doc.rust-lang.org/stable/std/collections/struct.LinkedList.html#method.cursor_front_mut